training/xgboost/structured/base/README.md

```diff
@@ -0,0 +1,150 @@
+
+## Explaining Key Elements
+
+In this section, we'll highlight the main elements of this sample.
+
+### [model.py](trainer/model.py)
+
+In this sample, we simply create an instance of `XGBClassifier` estimator and return it.
+
+### [metadata.py](trainer/metadata.py)
+
+We define which features should be used for training. We also define what the target is.
+
+### [train-local.sh](./scripts/train-local.sh)
+
+The command to run the training job locally is this:
+
+```bash
+gcloud ai-platform local train \
+        --module-name=trainer.task \
+        --package-path=${PACKAGE_PATH} \
+        --job-dir=${MODEL_DIR} \
+        -- \
+        --log-level DEBUG \
+        --input=${TAXI_TRAIN_SMALL} \
+        --n-estimators=20 \
+        --max-depth=3
+```
+
+* `--job-dir`: path for the output artifacts, e.g. the model object
+* `module-name` is the name of the Python file inside the package which runs the training job
+* `package-path` determines where the training Python package is.
+* `--` this is just a separator. Anyhing after this will be passed to the training job as input argument.
+* `log-level` sets the python logger level to DEBUG for this script (default is INFO)
+* `--input`: path to the input dataset
+* `--n-estimators` and `--max-depth`: will be passed to the estimator as hyperparameters
+
+
+### [train-cloud.sh](./scripts/train-cloud.sh)
+
+To submit a training job to AI Platform, the main command is:
+
+```bash
+gcloud ai-platform jobs submit training ${JOB_NAME} \
+        --job-dir=${MODEL_DIR} \
+        --runtime-version=${RUNTIME_VERSION} \
+        --region=${REGION} \
+        --scale-tier=${TIER} \
+        --module-name=trainer.task \
+        --package-path=${PACKAGE_PATH}  \
+        --python-version=${PYTHON_VERSION} \
+        --stream-logs \
+        -- \
+        --input=${GCS_TAXI_TRAIN_BIG} \
+        --n-estimators=20 \
+        --max-depth=3
+```
+
+* `${JOB_NAME}` is a unique name for each job. We create one with a timestamp to make it unique each time.
+* `runtime-version`: which runtime to use. See [this](https://cloud.google.com/ml-engine/docs/tensorflow/runtime-version-list) for more information.
+* `scale-tier` is to choose the tier. For this sample, we use BASIC. However, if you need
+to use accelerators for instance, or do a distributed training, you will need a different tier.
+* `region`: which region to run the training job in.
+* `stream-logs`: streams the logs until the job finishes. 
+
+## Clean Up
+If you were able to run [train-cloud.sh](./scripts/train-cloud.sh) successfully, you have
+created and stored some files in your GCS bucket. You may simply remove them by running
+
+```bash
+source ./scripts/cleanup.sh
+```
+
+## What's Next
+
+In this sample, we trained a simple classifier with xgboost.
+To see how to deploy the model to AI Platform and use it to make predictions,
+please continue with [this sample](../../../../prediction/xgboost/structured/base).
```

Comment:
that link 404s for me.

Reply:
Yes, I am going to implement it after this sample. It will not be 404 in a few days.

Reply:
Might be worth removing for now and adding later, but idk if someone is really going to stumble onto this repo.

Reply:
Right, but someone not familiar with GCP won't know what that means.

Reply:
The repo is private. This will be a non-issue by next week. I will keep it.   :)

Reply:
It isn't private anymore

Reply:
Oh really? I did not know. Let me remove it then.

Reply:
I commented it out (invisible in the readme).